sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -173,7 +200,33 @@ class GpuSortAggregateMeta(
         case "partial" => if (hashAggMode.contains(Final) || hashAggMode.contains(Complete)) {
           // replacing only Partial hash aggregates, so a Final or Commplete one should not replace
           willNotWorkOnGpu("Replacing Final or Complete hash aggregates disabled")
-         }
+        }
+          // In partial mode, if there are non distinct functions and multiple distinct functions
+          // non distinct functions are computed using First operator. The final result would be
+          // incorrect for non distinct functions for partition size > 1. Reason for this is - if
+          // the first batch computed and sent to CPU doesn't contain all the rows required to
+          // compute non distinct function(s), then Spark would consider that value as final result
+          // (due to First)Falling back to CPU for this special case.
+          if (agg.aggregateExpressions.exists(e => e.aggregateFunction.isInstanceOf[First])) {
+            agg.aggregateExpressions.foreach(e => {
+              // Check if there is an `If` within `First`. This is included in the plan only when
+              // multiple distinct is present in the query.  We fall back to CPU in this case when
+              // the references of `If` is an aggregate.
+              if (e.aggregateFunction.isInstanceOf[First]) {
+                val t = e.aggregateFunction.asInstanceOf[First].child.isInstanceOf[If]
+                if (t) {
+                  // Get the aggregate from references of `If`
+                  val aggRefs = e.aggregateFunction.references.tail.head
+                  val aggType = validateAggregate(aggRefs)
+                  if (aggType) {
```

Comment:
Nit: variable name is not indicative of what it holds, also can we add documentation above as to why we cant do something like `isDistinct` check here?